### DISCUSSION TITLE
analog sensor reading using pybricks

### DISCUSSION THREAD

**Discussion by jefromson** [pybricks/support#533] [Q&A]
Hi there,

I am attempting to get an analog sensor reading using pybricks with a legoEV3 brick and custom ultrasonic sensor and phototransistor light sensors.

I am using this code:
```
ultrasonic = AnalogSensor(Port.S1)
ultrasonic.active()
reading = ultrasonic.voltage()
```

When I log this value I get constant readings of 400 from the ultrasonic sensor and 415 from the light sensors regardless of how I move them around and what they are looking at. I would expect this value to change as they move around. I do know that the sensors are functional as I can run a matlab script with ev3 extension and read raw values from them successfully.

I should note that the way these sensors are set up the ev3 brick is being tricked into thinking that they are touch tensors so using the pybricks "TouchSensor" method I can return "False" values as well. I am unsure if this is affecting how pybricks is reading from them.

If anyone has any ideas I would very much appreciate them.

Thanks,
John

**Comment by laurensvalk** [comment]
After running this script, what do you see on the EV3 menu if you go to devices --> sensors?

Also, to which pin (colors) is the analog sensor wired?

**Reply by jefromson** [reply]
I am not with it now currently but if I remember correctly it was registering in the sensors menu as a touch sensor and giving me a "0".

The sensor is wired as:

red/yellow = ground
green = input/Vcc
blue = output
white goes through a 910Ohm resistor to ground
black is not connected

![image](https://user-images.githubusercontent.com/66144986/142435071-975752ac-a0a4-489e-8878-b6e2a70f5983.png)

**Reply by laurensvalk** [reply]
Thanks for getting back.

>  I remember correctly it was registering in the sensors menu as a touch sensor and giving me a "0".

That would be expected indeed on bootup, but it should change once you run this script.

This should change the port to analog, so it would be good to confirm that.

**Reply by jefromson** [reply]
Thank you for your prompt responses and assistance. My teammate ended up getting this to work using a custom sensor class calling the ev3devsensor method and feeding in the necessary mode. Seems to be working well!

Thanks again.
